File: Practica 3/python/Pila.py
class Pila:
    def __init__(self,n):
        self.arreglo=[]
        self.top=-1
        self.n=n
    def push(self,e):
        if self.isFull():
            print("Pila Llena: Push() no posible")
        else:
            self.top+=1
            self.arreglo.append(e)
            print(f"Push(): [{e}] añadido a la pila")
    def pop(self):
        if self.isEmpty():
            print("Pila vacía: Pop() no posible")
        else:
            e=self.arreglo[self.top]
            self.top-=1
            print(f"pop(): [{e}] eliminado de la pila")
            return e

    def peek(self):
        if self.isEmpty():
            print("Pila vacía: Peek() no posible")
        else:
            return self.arreglo[self.top]
    def isEmpty(self):
        return self.top==-1
    def isFull(self):
        return self.top==self.n-1
    
    
pila = Pila(3)

pila.push(10)
pila.push(20)
pila.push(30)
pila.push(40)

print(pila.peek())

pila.pop()
pila.pop()
pila.pop()
pila.pop()

File: Practica 3/java/Pila.java
public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.arreglo = new long[n];
        this.top = -1;
        this.n = n;
    }

    public void push(long e) {
        if (!isFull()) {
            top++;
            arreglo[top] = e;
            System.out.printf("Elemento añadido: %d\n", e);
        } else {
            System.out.println("Pila llena: no se puede insertar");
        }
    }

    public long pop() {
        if (!isEmpty()) {
            long elemento = arreglo[top];
            top--;
            System.out.printf("Elemento eliminado: %d\n", elemento);
            return elemento;
        } else {
            System.out.println("Pila vacía: no se puede eliminar");
            return -1;
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[top];
        } else {
            System.out.println("Pila vacía: no se puede hacer peek");
            return -1;
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args) {
        Pila pila = new Pila(3);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        pila.push(40);
        System.out.println("Elemento en la cima: " + pila.peek());
        pila.pop();
        pila.pop();
        pila.pop();
        pila.pop();

    }
}